### PR TITLE
Cell community detection

### PR DESCRIPTION
**Pull request by vladimirkovacevic**
Cell community detection to be integrated into stereopy.

### CODE REVIEW

File: stereo/tools/ccd/utils.py
Comment (LyonLen) [inline stereo/tools/ccd/utils.py:1]: should not use site-package scanpy

File: stereo/tools/community_detection.py
Comment (LyonLen) [inline stereo/tools/community_detection.py:1]: should inherit from `AlgorithmBase` and  expose the main method

**Review by LyonLen — COMMENTED** [review]
I'm sorry to inform You that the PR-reviewed-result is rejected.

**Comment by LyonLen** [comment]
Forgive my bothering, a demo run script and samples will be needed. Thx~

File: stereo/tools/ccd/community_clustering_algorithm.py
[the remark this answers is not in the document]
Comment (vladimirkovacevic) [reply stereo/tools/ccd/community_clustering_algorithm.py:1]: @LyonLen Stereopy has some limitations when reading .h5ad, for example both functions in stereopy for reading .h5ad (read_stereo_h5ad and read_ann_h5ad) can't read the cell type annotations (available in anndata.obs or anndata.obsm) that is necessary for CCD. How should we proceed with this?

**Pull request merged**